File: Python/Problem_2125.py
from typing import List

#   Problem - 2125: https://leetcode.com/problems/number-of-laser-beams-in-a-bank/
#   Python Solution!
class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        lastRowCount = bank[0].count('1')
        res = 0

        for row in range(1, len(bank)):
            thisRowCount = bank[row].count('1')

            if thisRowCount:
                res += lastRowCount * thisRowCount
                lastRowCount = thisRowCount

        return res


if __name__=="__main__":
    sol = Solution()

    print(sol.numberOfBeams(["011001","000000","010100","001000"]))
    print(sol.numberOfBeams(["000","111","000"]))

File: Cpp/Problem_2125.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

// Problem - 2125: https://leetcode.com/problems/number-of-laser-beams-in-a-bank/
// C++ Solution!
class Solution {
public:
	int numberOfBeams(vector<string>& bank) {
		int lastRowCount = 0, thisRowCount = 0, res = 0;

		for (string row : bank) {
			for (char ch : row) {
				if (ch == '1') {
					thisRowCount++;
				}
			}

			if (thisRowCount) {
				res += lastRowCount * thisRowCount;
				lastRowCount = thisRowCount;
				thisRowCount = 0;
			}
		}

		return res;
	}
};

int main() {
	Solution sol;

	vector<string> input = { "011001", "000000", "010100", "001000" };
	cout << sol.numberOfBeams(input) << endl;
	input = { "000", "111", "000" };
	cout << sol.numberOfBeams(input) << endl;

	return 0;
}